test/spdm-responder-emu/src/main.rs

```diff
@@ -25,10 +36,104 @@ use spdm_emu::socket_io_transport::SocketIoTransport;
 use spdm_emu::spdm_emu::*;
 use spdmlib::{common, config, protocol::*, responder};
 
-fn process_socket_message(
-    stream: &mut TcpStream,
-    transport_encap: &mut dyn SpdmTransportEncap,
-    buffer: &[u8],
+use std::mem::size_of;
+
+use spin::Mutex;
+extern crate alloc;
+use alloc::sync::Arc;
+use core::ops::DerefMut;
+
+// the possible path to create max stack consumption is either handle_spdm_certificate or handle_spdm_measurement in session.
+// there is no harm to sum these two to be the target stack size preservation.
+const EMU_STACK_SIZE: usize = size_of::<spdmlib::common::SpdmContext>() +
+                            size_of::<SpdmCertChainData>() * (SPDM_MAX_SLOT_NUMBER + 1) + size_of::<SpdmCertChainBuffer>() * SPDM_MAX_SLOT_NUMBER + // SpdmProvisionInfo
+                            size_of::<SpdmCertChainData>() * (SPDM_MAX_SLOT_NUMBER + 1) + // SpdmPeerInfo
+                            size_of::<SocketIoTransport>() + // SpdmDeviceIo
+                            size_of::<PciDoeTransportEncap>() + // PciDoeTransportEncap
+                            size_of::<MctpTransportEncap>() + // SpdmTransportEncap
```

Comment:
please use MAX(PciDoeTransportEncap, MctpTransportEncap).
Also, we can define a new macro for transport required size.

Reply:
> please use MAX(PciDoeTransportEncap, MctpTransportEncap). Also, we can define a new macro for transport required size.

updated.